definitions/checks/katello_agent.rb

```diff
@@ -0,0 +1,39 @@
+require 'yaml'
+
+class Checks::CheckKatelloAgentEnabled < ForemanMaintain::Check
+  metadata do
+    label :check_katello_agent_enabled
+    description 'Check whether the katello-agent feature is enabled before upgrading'
+    tags :pre_upgrade
+
+    confine do
+      feature(:instance).downstream
```

Comment:
why is that confined to downstream?

also, should that run on proxies, or just on the server?

Reply:
Regarding whether it should be run on proxies, I guess it's not necessary, because the installer migration already disables the feature and the puppet modules already perform the cleanup. The purpose of the check is to be an informative courtesy reminder to users, so running the check on each proxy seems excessive.

My understanding of foreman_maintain is that it's not used much by upstream, so I originally restricted this check to run downstream only in case we want to directly link any downstream documentation. I could lift this restriction if it's desired.

Reply:
I agree, by the time the server is upgraded, katello-agent support is gone already and the proxies don't matter anymore.